php/ad-servers/class-ad-layers-dfp.php

```diff
@@ -300,11 +300,30 @@ public function get_settings_fields() {
 					'code' => new Fieldmanager_Textfield( __( 'Code', 'ad-layers' ) ),
 					'path_override' => new Fieldmanager_TextField( __( 'Custom Path Template', 'ad-layers' ) ),
 					'sizes' => new Fieldmanager_Group( array(
-						'limit' => 0,
+						'add_more_label' => __( 'Add Size', 'ad-layers' ),
+						'description' => __( 'Size cannot be 0x0.', 'ad-layers' ),
+						'group_is_empty' => function( $values ) {
+							if ( isset( $values['width'], $values['height'] ) ) {
+								if ( is_numeric( $values['width'] ) && is_numeric( $values['height'] ) ) {
+									// Disallow 0px X 0px for an ad unit size. This sometimes leads to an "unremovable" field in the group.
+									if ( empty( $values['width'] ) && empty( $values['height'] ) ) {
+										return true;
+									}
+
+									// Valid size, save the group.
+									return false;
+								} else {
+									return true;
```

Comment:
So if the values are set but one of them is not numeric, the group should be considered empty?

Reply:
I think that the numeric checks can be removed since these fields are being sanitized by `absint` anyway.